srcpkgs/passt/template

```diff
@@ -0,0 +1,21 @@
+# Template file for 'passt'
+pkgname=passt
+version=2023.02.27.c538ee8
+revision=1
+# upstream uses YYYY_MM_DD.COMMIT
+_version=${version/./_}
+_version=${_version/./_}
```

Comment:
Debian seems to use `0.0~gitYYYYMMDD`, Fedora uses `0^YYYYMMDD`, I think we should go with them. As those version was listed as official in `passt` homepage?

Reply:
By the way, Debian package versions are formed with this [uscan rule](https://salsa.debian.org/sbrivio/passt/-/blob/master/debian/watch#L2):

```
options="uversionmangle=s/_//g; s/^/0.0~git/"
```

and this is the [rpkg macro](https://passt.top/passt/tree/contrib/fedora/rpkg.macros#n18) for Fedora:

```
printf "0^%s.g%s" "$(date -uI -d "${__date}" | tr -d -)" "${__commit}"
```

those are just to comply with distribution guidelines, there's no particular reason why. Upstream versions (tags) simply uses `YYYY_MM_DD.SHORT_SHA` as you noted because I find it more readable.

>  Or are you mainly suggesting prepending a 0-Version number in case they start using proper version?

There are no plans to switch to semantic versioning or suchlike. The current agreement, for the foreseeable future, is that dates are more meaningful than version numbers, yet should be readable enough, and the SHA immediately and uniquely identifies tags. We have no branches or "stable" releases, so that's quite convenient.

But sure, prepending a `0` is probably not a bad idea, even though it's a bit less readable. If we ever switch to a different scheme, we could also start with a `3`.

Reply:
@sbrivio-rh thanks for checking in. I can live with the current format. It's just that we don't allow `-` and `_` in the version part, so we can unambiguously match it. And if no semantic (that's the correct word, not "proper" :) ) versioning is planned, I think it's best to keep the `.`, since they allow me to generate the format with built-ins.
Getting `musl` support would be nice though, but I failed to find the correct includes for `__bswap_constant_16` for example.

Reply:
> I think it's best to keep the ., since they allow me to generate the format with built-ins.

Sure, it looks a bit nicer too.

> Getting musl support would be nice though, but I failed to find the correct includes for __bswap_constant_16 for example.

I guess that needs something like [this](https://github.com/analogdevicesinc/libiio/commit/9c6c6a432a0cbe2832bc97a7eeddfb61f6b8b856) -- probably a good fit for `util.h`.

In general I'd say we should carry own implementations for fundamental things like that (we really shouldn't swap those values at runtime), and instead change whatever code we didn't write in a portable way by mistake.

Reply:
See also https://bugs.passt.top/show_bug.cgi?id=4#c1 (!)